### PR TITLE
Return json if basic auth fail

### PR DESCRIPTION
**Pull request by AndrewTorrance**
# Motivation and Context
If the basic auth credentials failed then frontstage would die in a heap. This was because it was looking for json that was not present

# What has changed
Used the flask-hhtpauth package . Decared a @auth.error_handler function which now returns json with the keys that frontstage expects so that a meaningful error is logged

# How to test?
Run tests

# Links
https://trello.com/c/XpfbLBNw

### CODE REVIEW

File: ras_rm_auth_service/basic_auth.py [hunk 1]
```diff
@@ -15,3 +15,8 @@ def get_pw(username):
     config_password = current_app.config['SECURITY_USER_PASSWORD']
     if username == config_username:
         return config_password
+
+
+@auth.error_handler
+def auth_error():
+    return jsonify({"detail": "Incorrect basic auth"})
```
Comment (necrophonic) [inline ras_rm_auth_service/basic_auth.py]: Be nice if this conformed to RFC7807 (https://tools.ietf.org/html/rfc7807).
tl;dr:
Should return a `status_code` and can optionally return other fields including `detail` (as it already has - but more expanded), `type` (a link to a page describing the problem - skip that one), and `title` (which is the short description)

Comment (AndrewTorrance) [reply ras_rm_auth_service/basic_auth.py]: It  already returns 401 handled by the framework ( and tested for ) . I like the idea of the error generator having title and detail, means that the source of the error describes it not the consumer . Will change frontstage to cope also
